cli/args/config_file.rs

```diff
@@ -428,6 +440,16 @@ pub struct LintConfig {
   pub report: Option<String>,
 }
 
+impl LintConfig {
+  pub fn extend_files(&self, files: &FilesConfig) -> Self {
```

Comment:
Instead of cloning, let's change this to either consume `self` (`with_files`) or mutate `&mut self` (`extend_files`). Usually `extend` methods mutate (https://doc.rust-lang.org/std/iter/trait.Extend.html#tymethod.extend)

Reply:
alright, i'll change the function to consume the self as you suggested, like
```rs
impl LintConfig {
  pub fn with_files(self, files: &FilesConfig) -> Self {
    self.files.extend(files);
    self
  }
}
```